.github/workflows/build.yaml

```diff
@@ -0,0 +1,19 @@
+name: build
+
+on:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v*.*.*"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
You need to have the event name in the concurrency settings so that pushes to `main` don't cancel release builds. Let's fix wherever you copied this from, too.
```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
```

Reply:
`github.ref` should differentiate between branch and tag pushes per the docs below:

- https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

![image](https://github.com/rapidsai/rapids-build-backend/assets/7400326/6a0987cc-6d4a-42b5-8a8b-eaf96843c56d)

Reply:
Hmm. We added this across RAPIDS due to your request: https://github.com/rapidsai/ops/issues/2932

Reply:
That was a little different. We added that because the `build.yaml` workflows are triggered by `push` events (for PR merges) and `workflow_dispatch` events for nightly builds:

- https://github.com/rapidsai/rmm/blob/a98931b91ae8109f478469fd4eec388cf86b81fb/.github/workflows/build.yaml#L4-L9

Since we didn't have `event_name` in the concurrency group, there were scenarios where a PR merge would cause a nightly build to get canceled.

This workflow looks like it's only triggered by `push` events.

We could keep `github.event_name` if we expect this workflow to add additional events in the future, but as of now it's extraneous.

Reply:
I see. I'm slightly in favor of adding this even though it's extraneous. It's one fewer thing to think about and avoids the question of "why does this differ from the rest of RAPIDS?"

Reply:
fine by me